File: 03-backend/app/schemas/file.py
from typing import List, Optional
from pydantic import BaseModel

class FileNodeResponse(BaseModel):
    name: str
    path: str
    type: str  # "file" | "directory"
    size: Optional[int] = None
    extension: Optional[str] = None
    children: Optional[List["FileNodeResponse"]] = None

class FileTreeResponse(BaseModel):
    files: List[FileNodeResponse]

class FileContentResponse(BaseModel):
    name: str
    path: str
    content: str
    language: Optional[str] = None
    size: int

File: 02-frontend/src/types/file.ts
export interface FileNode {
  name: string;
  path: string;
  type: 'file' | 'directory';
  size?: number;
  extension?: string;
  children?: FileNode[];
}

export interface FileContent {
  path: string;
  name: string;
  content: string;
  language?: string;
  size: number;
}
